.pre-commit-config.yaml

```diff
@@ -1,39 +1,59 @@
 exclude: git-clang-format
 repos:
+  - repo: https://github.com/pre-commit/pre-commit-hooks
+    rev: v4.0.1
+    hooks:
+      - id: trailing-whitespace
+      - id: end-of-file-fixer
+      - id: fix-byte-order-marker
+      - id: check-builtin-literals
+      - id: check-vcs-permalinks
+      - id: check-ast
+      - id: check-yaml
+      - id: check-toml
+      - id: check-added-large-files
+      - id: check-merge-conflict
+      - id: mixed-line-ending
+      - id: check-executables-have-shebangs
+      - id: debug-statements
+      - id: destroyed-symlinks
+      - id: detect-private-key
+      - id: forbid-new-submodules
+  - repo: https://github.com/pre-commit/pygrep-hooks
+    rev: v1.9.0
+    hooks:
+      - id: python-check-blanket-type-ignore
+      - id: python-use-type-annotations
+      - id: python-check-mock-methods
+      - id: python-check-blanket-noqa
+      - id: python-no-eval
+      - id: rst-backticks
+      - id: rst-directive-colons
+      - id: rst-inline-touching-normal
+      - id: text-unicode-replacement-char
+  - repo: https://github.com/PyCQA/isort
+    rev: 5.9.3
+    hooks:
+      - id: isort
   - repo: https://github.com/psf/black
     rev: 21.7b0
     hooks:
       - id: black
+  - repo: https://gitlab.com/PyCQA/flake8
+    rev: 3.9.2
+    hooks:
+      - id: flake8
+        exclude: |
+            (?x)^(
+                check_do_not_submit.py|
+
+            )$
   - repo: https://github.com/pre-commit/mirrors-mypy
     rev: v0.910
     hooks:
       - id: mypy
         args: [--strict]
-  - repo: https://github.com/psf/black
```

Comment:
Could you tidy up this diff so that it is easier to review?

You have almost the same checks as in the current project `HEAD` but you've reordered them.

It would be _much_ easier if the diff just showed what you have added and removed rather than what you have added, removed *and* moved.

Reply:
updated